### PR TITLE
Explain how to apply attributes to positional records

### LINKED ISSUE #24037: Record reference should include attribute target for positional records
Positional records can include the `field:` or `property:` attribute target on the primary constructor in order to apply attributes to the field or property generated from those parameters. (You can also apply the `parameter:` but that is the default).

In addition, this content should be linked from the following locations:

- Attributes that affect compilation, in the section on AttributeUsage
- JSON Serialization: One of the most common scenarios for using this will be applying JSON serialization attributes to the record's properties. Likely candidates would be the name, or the ignore attribute.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 20818c64-4ad8-fb15-5d79-e7f98614e9c4
* Version Independent ID: de1f383b-ef20-4776-57a8-04edae892166
* Content: [Records - C# reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record)
* Content Source: [docs/csharp/language-reference/builtin-types/record.md](https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/builtin-types/record.md)
* Product: **dotnet-csharp**
* Technology: **csharp-language-reference**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #24037 

Add a description and example for how to apply attributes to the field or property generated from the primary constructor on positional records. 

Then, link to that description from the locations where readers are mostly likely to need to add attributes to records, or where they are likely to look for that information.

### CODE REVIEW

File: docs/csharp/language-reference/attributes/general.md [hunk 1]
```diff
@@ -90,6 +90,8 @@ If <xref:System.AttributeUsageAttribute.Inherited> is `false`, then the attribut
 
 In this case `NonInheritedAttribute` isn't applied to `DClass` via inheritance.
 
+You can also use these keywords to specify where an attribute should be applied. For example, you can use the `field:` specifier to add an attribute to the backing field of an [auto-implemented property]../../programming-guide/classes-and-structs/properties.md#auto-implemented-properties). Or you can use the `field:`, `property:` or `param:` specifier to apply an attribute to any of the elements generated from a [positional record](../builtin-types/record.md#positional-syntax-for-property-definition).
```
Comment (tdykstra) [inline docs/csharp/language-reference/attributes/general.md:93]: Might be worth adding examples here.

```suggestion
You can also use these keywords to specify where an attribute should be applied. For example, you can use the `field:` specifier to add an attribute to the backing field of an [auto-implemented property](../../programming-guide/classes-and-structs/properties.md#auto-implemented-properties). Or you can use the `field:`, `property:` or `param:` specifier to apply an attribute to any of the elements generated from a positional record. For an example, see [Positional syntax for property definition](../builtin-types/record.md#positional-syntax-for-property-definition).
```

File: docs/csharp/language-reference/builtin-types/record.md [hunk 2]
```diff
@@ -46,6 +46,12 @@ When you use the positional syntax for property definition, the compiler creates
 * A primary constructor whose parameters match the positional parameters on the record declaration.
 * A `Deconstruct` method with an `out` parameter for each positional parameter provided in the record declaration. This method is provided only if there are two or more positional parameters. The method deconstructs properties defined by using positional syntax; it ignores properties that are defined by using standard property syntax.
 
+You may want to add attributes to any of these elements the compiler creates from the record definition. You can add a *target* to any atribute you apply to the positional record's properties. The following example applies the <xref:System.Text.Json.Serialization.JsonPropertyNameAttribute?displayProperty=nameWithType> to each property of the `Person` record. The `property:` target indicates that the attribute is applied to the compiler generated property. Other values are `field:` to apply the attribute to the field, and `param:` to apply the attribute to the parameter.
```
Comment (tdykstra) [inline docs/csharp/language-reference/builtin-types/record.md:49]:
```suggestion
You may want to add attributes to any of these elements the compiler creates from the record definition. You can add a *target* to any attribute you apply to the positional record's properties. The following example applies the <xref:System.Text.Json.Serialization.JsonPropertyNameAttribute?displayProperty=nameWithType> to each property of the `Person` record. The `property:` target indicates that the attribute is applied to the compiler generated property. Other values are `field:` to apply the attribute to the field, and `param:` to apply the attribute to the parameter.
```

File: docs/standard/serialization/system-text-json-immutability.md [hunk 3]
```diff
@@ -42,6 +42,8 @@ Records in C# 9 are also supported, as shown in the following example:
 
 :::code language="csharp" source="snippets/system-text-json-how-to-5-0/csharp/Records.cs":::
 
+You can apply any of the attributes to the property names, using the `property:` target on the attribute. For more information on positional records, see the article on [records](../../csharp/language-reference/builtin-types/record.md#positional-syntax-for-property-definition)
```
Comment (tdykstra) [inline docs/standard/serialization/system-text-json-immutability.md:45]:
```suggestion
You can apply any of the attributes to the property names, using the `property:` target on the attribute. For more information on positional records, see the article on [records](../../csharp/language-reference/builtin-types/record.md#positional-syntax-for-property-definition) in the C# language reference.
```

**Review by tdykstra — APPROVED** [review]
Looks great, a couple typos and suggestions.

File: docs/csharp/language-reference/builtin-types/snippets/shared/builtin-types.csproj [hunk 4]
```diff
@@ -9,4 +9,8 @@
     <Platforms>AnyCPU;x86</Platforms>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <DocumentationFile></DocumentationFile>
+  </PropertyGroup>
```
Comment (pkulikov) [inline docs/csharp/language-reference/builtin-types/snippets/shared/builtin-types.csproj:14]: @BillWagner what is the purpose of this addition to the project file, or this is accidental?

**Pull request merged**